pkg/credentialprovider/plugin/plugin.go

```diff
@@ -193,12 +207,27 @@ func (p *pluginProvider) Provide(image string) credentialprovider.DockerConfig {
 		return cachedConfig
 	}
 
-	response, err := p.plugin.ExecPlugin(context.Background(), image)
+	// ExecPlugin is wrapped in single flight to exec plugin once for concurrent same image request.
+	// The caveat here is we don't know cacheKeyType yet, so if cacheKeyType is registry/global and credentials saved in cache
```

Comment:
So in the event that there are multiple concurrent calls, the last successful call is the one that is cached right?

Reply:
yes, the disadvantage is just that for same credentials it could request multiple times. 
We could change the key for singleflight if `cacheKeyType` is known in advance or we were saving it in config file.